nntrainer/tensor/tensor_pool.h

```diff
@@ -340,18 +299,35 @@ class TensorPool {
                         unsigned int offset);
 
 private:
+  /**
+   * @brief Source tensor detailed specification
+   *
+   */
+  struct SourceDetails {
+    unsigned int token;                   /**< memory token */
+    TensorLifespan lifespan;              /**< life span of the tensor */
+    std::vector<unsigned int> exec_order; /**< exec order */
+    std::vector<unsigned int>
```

Comment:
does SourceDetails need to store all its dependents index? In my understanding, this is stored so that source can initialize all its dependents. It would be easier to rather each dependent finalize themselves from the source once the source has been finalized.

Reply:
The main goal is to enable `It would be easier to rather each dependent finalize themselves from the source`  

You can think this as a pub sub pattern.

Publisher is the source tensor, subscriber is dependents. If the source have to notify something, in my understanding, the source have to keep list of subscribers.

Before this patch, there was no such mechanism to enable this, so there was sthlike `updateAll()` asking the user of tensor pool to call periodically.

If you'd have better idea, I'd like to listen :)

Reply:
In my understanding, the design of tensor_pool is such that all the tensors are allocated at once and freed at once. So, all the tensors will be updated at once.

If this is regarding updating the tensors which are dependent on external tensors, then it's best to separate out the list of external tensors inside tensor_pool. We need to start considering the overhead of storing our metadata per tensor as it is becoming heavier over time.

Please add a corresponding TODO and lets go forward with this for now.